src/js/engagement_view/src/components/graphDisplay/graphLayout/mergeGraphs.tsx

```diff
@@ -124,3 +108,46 @@ export const mergeGraphs = (
         return null;
     }
 };
+
+const mergeSummarizedLink = (oldLink: SummaryLink, newLink: SummaryLink): [boolean, SummaryLink] => {
+    const allInnerLinks = [...oldLink.innerLinks, ...newLink.innerLinks];
+    const summarizedLink = summarizeLinks(allInnerLinks)[0];
+    const updated = summarizedLink.innerLinks.length !== oldLink.innerLinks.length;
+
+    return [updated, summarizedLink];
+};
+
+const mergeSummarizedLinks = (oldLinks: SummaryLink[], newLinks: SummaryLink[]): [boolean, SummaryLink[]] => {
+    let updated = false;
+    const mergedLinks: Map<string, SummaryLink> = new Map();
+
+    for (const link of oldLinks) {
+        const key = [link.source, link.target];
+        key.sort();
+
+        const sKey = String(key[0]) + String(key[1]);
```

Comment:
Might be a good idea to move this key gen stuff into a function like `linkKey`, with a comment explaining that we're normalizing the order because links are bidirectional.